modules/panels.nix

```diff
@@ -231,12 +240,23 @@ in
                 desktop.currentConfigGroup = Array("Wallpaper", "org.kde.color", "General");
                 desktop.writeConfig("Color", "${cfg.workspace.wallpaperPlainColor}");
             }
+          '' else "");
+          desktopFolderSettings = (if anyDesktopFolderSettingsSet then ''
+            // Desktop folder settings
+            let allDesktops = desktops();
+            for (const desktop of allDesktops) {
+                desktop.currentConfigGroup = ["General"];
+                ${lib.optionalString (cfg.workspace.desktop.icons.arrangement == "topToBottom") ''desktop.writeConfig("arrangement", 1);''}
+                ${stringIfNotNull cfg.workspace.desktop.icons.size ''desktop.writeConfig("iconSize", ${builtins.toString cfg.workspace.desktop.icons.size});''}
+                ${lib.optionalString (!cfg.workspace.desktop.icons.folderPreviewPopups) ''desktop.writeConfig("popups", false);''}
+                ${stringIfNotNull cfg.workspace.desktop.icons.previewPlugins ''desktop.writeConfig("previewPlugins", "${lib.strings.concatStringsSep "," cfg.workspace.desktop.icons.previewPlugins}");''}
+              }
           '' else ""
           );
```

Comment:
I think it would be better to move this code into a separate script. I think you could instead put this into something like:
```nix
programs.plasma.startup.desktopScript."set_desktop_folder_settings" = (lib.mkIf anyDesktopFolderSettingsSet {
  text = ''
    {{SCRIPT_CONTENT_HERE}}
  '';
  priority = 3;
});
```
Where you would also move `anyDesktopFolderSettingsSet` to the `workspace.nix` of course.

The reason why the wallpaper and panels had to be put in the same file was that when they were in separate scripts and the panel was updated, the wallpaper script also had to be run for the wallpaper to be set (when `overrideConfig` was disabled). I'm not sure if this is the case here (but it might be). In any case though the plan is to split these into separate scripts again soon, so we might as well start here. I have a better workaround now if the same errors occur here.

Reply:
Thanks, I tried to change this in f28bd06, but haven’t had time to check if it works. I’d also like to test if the separate script problem arises here, but I don’t know how. If I understand correctly the wallpaper got lost on the next login after changing the panels, as the wallpaper script isn’t re-run due to being unchanged while the panel script is, which also deleted the `plasma-org.kde.plasma.desktop-appletsrc` and the wallpaper settings with it. In that case I just need to change some panel settings, log off and in again and check if the desktop folder settings are still there? Or am I wrong here?